src/Button/index.js

```diff
@@ -0,0 +1,271 @@
+   }
+
+   return buttonStyle
+})
+
+// ============================================================================
+// Component
+// ============================================================================
+
+export default class Button extends React.Component {
+   static propTypes = {
+      borderRadius: OIOResponsiveObjectPropType,
+      children: PropTypes.node,
+      className: PropTypes.string,
+      color: OIOResponsiveObjectPropType,
+      fontFamily: OIOResponsiveObjectPropType,
+      mode: PropTypes.oneOf(['disabled', 'normal', 'pulsing', 'loading']),
+      name: PropTypes.node,
+      onClick: PropTypes.func,
+      outline: PropTypes.bool,
+      padding: OIOResponsiveObjectPropType,
+      rounded: PropTypes.bool,
+      size: OIOResponsiveObjectPropType,
+      style: PropTypes.object,
+      tagName: PropTypes.oneOf(['button', 'div', 'span']),
+      textColor: OIOResponsiveObjectPropType,
+      textSize: OIOResponsiveObjectPropType,
+      textTransform: OIOResponsiveObjectPropType,
+      textWeight: OIOResponsiveObjectPropType,
+      type: PropTypes.oneOf(['button', 'clear', 'submit']),
+      width: OIOResponsiveObjectPropType
+   }
+
+   static defaultProps = {
+      borderRadius: r`4px`,
+      children: undefined,
+      mode: 'normal',
+      onClick: () => {},
+      outline: false,
+      rounded: false,
+      size: r`md`,
+      style: {},
+      tagName: 'button',
+      textColor: r`#fff`,
+      textSize: undefined,
+      textTransform: r`none`,
+      textWeight: r`medium`,
+      type: 'button',
+      width: r`auto`
+   }
+
+   render() {
+      const {
+         children, className, mode, name, tagName, type,
+         textColor, textSize, textTransform, textWeight, onClick,
+         backgroundColor, border, borderRadius, color, fontFamily,
+         height, padding, minWidth, width,
+         hoverBackgroundColor, hoverBorder
+      } = this.props
+
+      /* eslint-disable object-property-newline */
+      const responsiveStyles = generateResponsiveStyles({
+         backgroundColor, border, borderRadius, color, fontFamily,
+         height, padding, minWidth, width
+      })
+
+      const hoverResponsiveStyles = generateResponsiveStyles({
+         backgroundColor: hoverBackgroundColor,
+         border: hoverBorder
+      })
+      /* eslint-enable */
+
+      const textStyle = {}
+      const buttonStyle = {
+         alignItems: 'center',
+         cursor: 'pointer',
+         display: 'inline-flex',
+         justifyContent: 'center',
+         outline: 'none',
+         position: 'relative',
+         transition: '200ms',
+         ...this.props.style,
+         ...responsiveStyles
+      }
+
+      // ====================================================
+      // Element
+      // ====================================================
+
+      // Abilty to use custom tag (div, button, span)
+      const ButtonElement = tagName
+      const conditionalButtonProps = {}
```

Comment:
Any reason for the prefix `conditional`? I guess I ask because there are a ton of variables in the codebase that are set inside `if` or `else`, but do not have that prefix. Also if something is set in `else` does that make it conditional? Or if it is set in a ternary?